fml/src/main/java/net/minecraftforge/fml/common/discovery/JarDiscoverer.java

```diff
@@ -42,7 +43,7 @@
         {
             jar = new JarFile(candidate.getModContainer());
 
-            if (jar.getManifest()!=null && (jar.getManifest().getMainAttributes().get("FMLCorePlugin") != null || jar.getManifest().getMainAttributes().get("TweakClass") != null))
+            if (jar.getManifest()!=null && (jar.getManifest().getMainAttributes().get("FMLCorePlugin") != null || (jar.getManifest().getMainAttributes().get("TweakClass") != null && jar.getManifest().getMainAttributes().get("TweakName") == null)))
```

Comment:
You don't need this check. If something has a `TweakName`, it probably has a `TweakClass`. If it has a name, but no class, it won't be added anyway.